src/main/java/org/folio/list/services/export/CsvCreator.java

```diff
@@ -61,8 +66,10 @@ public ExportLocalStorage createAndUploadCSV(ExportDetails exportDetails, String
       if (batchNumber % 10 == 0) {
         checkIfExportCancelled(list.getId(), exportDetails.getExportId());
 
-        //Skip the first batch since we haven't generated any content yet.
-        if (batchNumber != 0) {
+        //Skip the first batch since we haven't generated any content yet and do not upload if file size less than 5 mb
+        File multiPartFile = new File(localStorage.getAbsolutePath());
+        long bytes = FileUtils.sizeOf(multiPartFile);
+        if (batchNumber != 0 && bytes > MINIMAL_PART_SIZE) {
```

Comment:
I could definitely be misunderstanding something here, but my concern with this is: While we are now checking the file size to make sure the upload doesn't fail, it doesn't look like we do anything if the number of bytes is less than the minimum size. Does that mean we're going to lose the last piece of the upload?

Reply:
It is anyway at line 88, the final upload part, so we do not lose any data due to its smaller size.